test/extension/config/default/manager.yaml

```diff
@@ -2,24 +2,24 @@
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: test-extension
+  name: manager
 spec:
   selector:
     matchLabels:
-      app: test-extension
+      app: test-extension-manager
```

Comment:
Those labels seem to diverge from our other providers (they are just using "controller-manager").

Probably it's overwritten somehow with kustomize?

Might be worth diffing this folder against the one of the core provider for consistency

Should we also split this into default, manager, rbac, .. folders?

Initially I just wanted to keep it as simple as possible for the extension. But given it's now a provider I wonder if we should have a kustomize folder structure consistent with our other providers

Reply:
This PR introduces only changes required to make those manifest compatible with the clusterctl contract; making this more consistent with other providers/kubebuilder can be tackled as a follow up action if we deem it necessary, but I kind of agree that keeping it simple is probably better.